app/admin-tab/machine/controller.js

```diff
@@ -0,0 +1,128 @@
+        newDriver.externalId  = driver.id;
+      }
+
+      this.set('newMachine', this.get('userStore').createRecord(newDriver));
+
+      this.get('application').setProperties({
+        editMachineDriver: true,
+        originalModel: this.get('newMachine'),
+      });
+    },
+    addCatalogDriver: function(driver) {
+      this.get('store').request({url: this.get('app.catalogEndpoint')+'/templates/'+driver.id}).then((template) =>{
+        this.get('store').request({url: template.versionLinks[template.defaultVersion]}).then((driver) =>{
+          let newDriver = {
+            type            : 'machineDriver',
+            description     : (driver.description || null),
+            checksum        : (driver.files.checksum || null),
+            url             : driver.files.url.replace(/[\n\r]+/g, ''),
+            externalId      : driver.id,
+            activateOnCreate: true,
+          };
+          this.get('userStore').createRecord(newDriver).save().then((newDriver) => {
+            this.get('model.drivers').pushObject(newDriver);
+          });
+        });
+      });
+    },
+    upgradeDriver: function(driver, version/*, path*/) {
+      let templateVersion = version;
+
+      this.set('upgrading', true);
+      // find latest version of driver
+      this.get('store').request({url: this.get('app.catalogEndpoint')+'/templateversions/'+driver.externalId}).then((template) => {
+        this.get('store').request({url: template.upgradeVersionLinks[templateVersion]}).then((item) => {
+          driver.setProperties({
+            url      : item.files.url,
+            checksum : item.files.checksum,
+            uiUrl    : item.files.uiUrl,
+          });
+          driver.save().then(() => {
+            this.set('upgrading', false);
+          });
+        });
+      });
+    }
+  },
+
+  sortableContent: Ember.computed('model.drivers.@each', 'model.catalogDrivers.@each', function() {
+    // possibly add some search here
+    let cDrivers   = this.get('model.catalogDrivers.catalog');
+    let drivers    = this.get('model.drivers.content');
+    let newContent = [];
+
+    cDrivers.forEach((cDriver) => {
+      let check = drivers.find((driver) =>{
+
+        if (driver.externalId && C.REMOVEDISH_STATES.indexOf(driver.get('state')) === -1) {
+
+          let extId = driver.externalId.split(':');
+          extId     = extId.slice(0, extId.length - 1).join(':');
+
+          if (cDriver.id === extId) {
+            this.get('store').request({url: this.get('app.catalogEndpoint')+'/templateversions/'+ driver.externalId}).then((upgradeInfo) => {
+              if (upgradeInfo.id === driver.externalId) {
+                return false;
+              }
+              if (upgradeInfo.upgradeVersionLinks && Object.keys(upgradeInfo.upgradeVersionLinks).length) {
+                driver.set('upgradeAvailable', true);
+                driver.set('upgradeVersionLinks', upgradeInfo.upgradeVersionLinks);
```

Comment:
`upgradeAvailable` seems redundant, since you have (or don't) some `upgradeVersionLinks`